main.cpp

```diff
@@ -0,0 +1,101 @@
+/* WiFi Example
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbed.h"
+#include "TCPSocket.h"
+
+#if TARGET_UBLOX_EVK_ODIN_W2
+    #include "OdinWiFiInterface.h"
+    OdinWiFiInterface wifi;
+#else
+    #include "ESP8266Interface.h"
+    ESP8266Interface wifi(D1, D0);
+#endif
+
+#define STR(x) STR2(x)
+#define STR2(x) #x
+
+// Socket demo
+void http_demo(NetworkInterface *net) {
+    TCPSocket socket;
+
+    printf("Sending HTTP request to www.arm.com...\r\n");
+
+    // Open a socket on the network interface, and create a TCP connection to www.arm.com
+    socket.open(net);
+    socket.connect("www.arm.com", 80);
+
+    // Send a simple http request
+    char sbuffer[] = "GET / HTTP/1.1\r\nHost: www.arm.com\r\n\r\n";
+    int scount = socket.send(sbuffer, sizeof sbuffer);
+    printf("sent %d [%.*s]\r\n", scount, strstr(sbuffer, "\r\n")-sbuffer, sbuffer);
+
+    // Recieve a simple http response and print out the response line
+    char rbuffer[64];
+    int rcount = socket.recv(rbuffer, sizeof rbuffer);
+    printf("recv %d [%.*s]\r\n", rcount, strstr(rbuffer, "\r\n")-rbuffer, rbuffer);
+
+    // Close the socket to return its memory and bring down the network interface
+    socket.close();
+}
+
+void scan_demo(WiFiInterface *wifi)
```

Comment:
Thought: Since the `scan_demo` is the interesting new functionality and is called before `http_demo` chronologically, should we put `scan_demo` before `http_demo` in the source code?